### PR TITLE
AS-655: Remove PerRequest pattern

### PR DESCRIPTION
**Pull request by MatthewBemis**
This had been sitting dormant in a PR [here](https://github.com/broadinstitute/rawls/pull/1408) waiting until PPW was out (to avoid causing extra merge conflict work for CI team). Easiest way to revive this work was to just create a new branch.

I am thinking the best way to review this will be to have 2 reviewers as usual but also make sure to give Batch and CI the opportunity to ask themselves "do we care about this change?" and peek if they feel like it's necessary

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 1]
```diff
@@ -495,19 +472,17 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
...
-  def adminDeleteRefreshToken(rawlsUserRef: RawlsUserRef): Future[PerRequestMessage] = {
+  def adminDeleteRefreshToken(rawlsUserRef: RawlsUserRef): Future[Unit] = {
     asFCAdmin {
-      deleteRefreshTokenInternal(rawlsUserRef).map(_ => RequestComplete(StatusCodes.OK))
+      deleteRefreshTokenInternal(rawlsUserRef)
     }
   }
 
-  def updateBillingProjectBillingAccount(billingProjectName: RawlsBillingProjectName, updateAccountRequest: UpdateRawlsBillingAccountRequest): Future[PerRequestMessage] = {
+  def updateBillingProjectBillingAccount(billingProjectName: RawlsBillingProjectName, updateAccountRequest: UpdateRawlsBillingAccountRequest): Future[Option[RawlsBillingProjectResponse]] = {
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:485]: BillingApiServiceV2, which calls this, needs a conditional to return `OK`/`NotFound` based on whether or not this returns `Some`/`None`

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 2]
```diff
@@ -521,13 +496,13 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
-  def deleteBillingAccount(billingProjectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+  def deleteBillingAccount(billingProjectName: RawlsBillingProjectName): Future[Option[RawlsBillingProjectResponse]] = {
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:499]: BillingApiServiceV2, which calls this, needs a conditional to return OK/NotFound based on whether or not this returns Some/None

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala [hunk 3]
```diff
@@ -1566,6 +1566,56 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
...
+  it should "return 403 for a writer testing the write action on a locked workspace" in withTestDataApiServicesAndUser("writer-access") { services =>
+    Get(s"${testData.workspaceLocked.path}/checkIamActionWithLock/write") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Forbidden) {
+          status
+        }
+      }
+  }
+
```
Comment (davidangb) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala:1618]: I'm curious - are you just adding coverage here? Was there a specific issue that needed testing?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala [hunk 4]
```diff
@@ -30,46 +34,51 @@ trait AdminApiService extends UserInfoDirectives {
...
         entity(as[Map[String, String]]) { ownerInfo =>
-          complete { userServiceConstructor(userInfo).adminUnregisterBillingProjectWithOwnerInfo(RawlsBillingProjectName(projectName), ownerInfo) }
+          complete { userServiceConstructor(userInfo).adminUnregisterBillingProjectWithOwnerInfo(RawlsBillingProjectName(projectName), ownerInfo).map(_ => StatusCodes.NoContent) }
         }
       }
     } ~
     path("admin" / "submissions") {
       get {
-        complete { workspaceServiceConstructor(userInfo).adminListAllActiveSubmissions() }
+        complete { workspaceServiceConstructor(userInfo).adminListAllActiveSubmissions().map(StatusCodes.OK -> _) }
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala]: this applies in a lot of places, unfortunately: isn't `.map(StatusCodes.OK -> _)` extraneous within a `complete`?

```
complete { workspaceServiceConstructor(userInfo).adminListAllActiveSubmissions().map(StatusCodes.OK -> _) }
```
could just be:
```
complete { workspaceServiceConstructor(userInfo).adminListAllActiveSubmissions() }
```
unless I'm missing something - you are closer to this than I am!

I'd be fine merging this PR and then cleaning that up as a follow-on

**Review by davidangb — COMMENTED** [review]
There are two places where I believe this PR changes the response codes. I also have one comment about potentially extraneous code. Other than that, looks great, and wow this is a big one!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala (hunk 4 above)
Comment (MatthewBemis) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala]: You're totally right- I think it's worth fixing now!

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala (hunk 3 above)
Comment (MatthewBemis) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala:1618]: This is restoring some coverage that was lost as a result of the changes in this PR. In the `ServiceSpec` we were testing response codes and response data. Now that the service layer isn't returning response codes anymore (except in the case of some exceptions), I needed to add that coverage in the `ApiSpec`

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala (hunk 1 above)
Comment (MatthewBemis) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:485]: Good catch!

**Comment by MatthewBemis** [comment]
jenkins retest

**Review by davidangb — APPROVED** [review]
lgtm, thanks for the tweaks!

**Review by ansingh7115 — APPROVED** [review]
Can't find anything to complain about! This is great!

**Pull request merged**